nio/events/ephemeral.py

```diff
@@ -84,3 +86,38 @@ class TypingNoticeEvent(EphemeralEvent):
     @verify_or_none(Schemas.m_typing)
     def from_dict(cls, parsed_dict):
         return cls(parsed_dict["content"]["user_ids"])
+
+
+@attr.s
+class ReadReceiptEvent(EphemeralEvent):
+    """Informs the client of changes in the newest events seen by users.
+
+    Each ReadReceiptEvent can contain multiple event_ids, each with many users
+    who have seen that event most recently.
+
+    Attributes:
+        receipts (dict): A dictionary mapping event_ids as keys to a list of
+            receipts. Each receipt is a dictionary with a `receipt_type`,
+            `user_id` and `timestamp`.
+    """
+
+    receipts = attr.ib(type=Dict)
+
+    @classmethod
+    @verify_or_none(Schemas.m_read)
+    def from_dict(cls, parsed_dict):
+        event_receipts = {}
+
+        for event_id, event in parsed_dict["content"].items():
+            receipts = []
+            for receipt_type, receipt in event.items():
+                for user_id, user in receipt.items():
+                    receipts.append({
+                        "receipt_type": receipt_type,
```

Comment:
I think we should create a class for the receipts or at least somewhere document what they contain. A class allows completion to work and would force us to document stuff properly.

Reply:
I'll add a new `ReadReceipt` class and change the `receipts` type from `attr.ib(type=Dict)` to `attr.ib(type=Dict[str, ReadReceipt])`; is that right?

Reply:
Yes.